numpy_financial/_financial.py

```diff
@@ -281,21 +281,25 @@ def nper(rate, pmt, pv, fv=0, when='end'):
 
     """
     when = _convert_when(when)
-    (rate, pmt, pv, fv, when) = map(np.asarray, [rate, pmt, pv, fv, when])
+    rate, pmt, pv, fv, when = np.broadcast_arrays(rate, pmt, pv, fv, when)
+    nper_array = np.empty_like(rate, dtype=np.float64)
```

Comment:
A quick comment for any other reviewers; unlike most other financial functions ``nper`` doesn't support ``Decimal`` inputs so having ``dtype=np.float64`` as the default won't cause an issue.